libs/tokens/src/pure/TokenLogo/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { atom, useAtom } from 'jotai'
+import { useMemo } from 'react'
+
+import { cowprotocolTokenUrl, NATIVE_CURRENCY_BUY_ADDRESS, TokenWithLogo } from '@cowprotocol/common-const'
+import { uriToHttp } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Currency, NativeCurrency } from '@uniswap/sdk-core'
+
+import { Slash } from 'react-feather'
+import styled from 'styled-components/macro'
+
+import { getTokenLogoUrls } from '../../utils/getTokenLogoUrls'
+
+const invalidUrlsAtom = atom<{ [url: string]: boolean }>({})
+
+const TokenLogoWrapper = styled.div`
+  display: inline-block;
+  background: var(--cow-container-bg-01);
+  border-radius: 50%;
+`
+
+export interface TokenLogoProps {
+  token?: TokenWithLogo | Currency | null
+  logoURI?: string
+  className?: string
+  size?: number
+}
+
+export function TokenLogo({ logoURI, token, className, size = 36 }: TokenLogoProps) {
+  const [invalidUrls, setInvalidUrls] = useAtom(invalidUrlsAtom)
+
+  const urls = useMemo(() => {
+    // TODO: get rid of Currency usage and remove type casting
+    if (token) {
+      if (token instanceof NativeCurrency) {
+        return [cowprotocolTokenUrl(NATIVE_CURRENCY_BUY_ADDRESS.toLowerCase(), token.chainId as SupportedChainId)]
+      }
+
+      return getTokenLogoUrls(token as TokenWithLogo)
+    }
+
+    return logoURI ? uriToHttp(logoURI) : []
```

Comment:
Is there any case where `token` is not set but `logoURI` is?

Reply:
Actually, the only case in `TokenListDetails`:

```
<TokenLogo logoURI={list.logoURI} size={36} />
```

I use `TokenLogo` component to display a list logo. Probably it's wrong, but I was too lazy to create another component :) I'm ok to create another component for it, just tell me

Reply:
I rather another component, because then we remove the ambiguity of the optional token for the TokenLogo